### PR TITLE
Add documentation for the door setup

### PR DESCRIPTION
**Pull request by SohamG**
## Description 

Add documentation of the newly refurbished door controller setup. 

## Issues

Discord bot is jank. Need to polish and upload source code.

## Checklist

- [X] I've labeled the PR appropriately.
- [X] I've have built the website and verified that my changes work.
- [X] I've linked the appropriate issues and related PRs.

### CODE REVIEW

File: docs/kb/office-door.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+- This shell script calls the EVDEV Driver with its output piped to `new_card_reader.sh`
+- Systemd messes up I/O redirection so the intermediary script is required.
+
+## Discord Bot
+
+- Systemd system service called discordbot.service.
+- Simple python script that uses `py-cord`.
+- Virtual Env in `/home/pi/doorkeeper-controller/botvenv`
+
+## Wishlist
```
Comment (bmiddha) [inline docs/kb/office-door.md:47]: nit: wish list is 2 words.
the wish list should also be moved to an issue in the code repo.

File: docs/kb/office-door.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+### Shell script
+
+- This is the legacy card_reader.sh (now called `new_card_reader.sh` script modified to have `?` as the delimiter.
+- UIC ID Cards have ? as the last character.
+- The output of the EVDEV Driver is piped to this shell script.
+- This script contains a list of UINs allowed to open the door.
+- TODO: Make it read allowed UINs from a file (BASH `mapfile` builtin).
+- If the card is of an allowed UIN, the python scirpt to open the door is called.
+
+### Deployment Information
```
Comment (bmiddha) [inline docs/kb/office-door.md:34]: The issue with keeping the deployment and implementation specific docs here is that they can *easily* get out of sync. It's better to keep this stuff in the repo and point to it from here instead.

We should document the existence of the project, what it's about, how it works with other projects, where to look at the code (implementation, deployment docs, etc.)

File: docs/kb/office-door.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## EVDEV driver
+
+- The card reader is basically the keyboard, presented in /dev/input/eventX.
+- The Rust program hijacks the input device to prevent spurious inputs to the OS.
+  (Technically, the raspberry pi would just be sitting at the TTY login prompt. Don't want
+  the input to also go there).
+- Key press and unpress events are processed with [libxkbcommon](https://xkbcommon.org/)
+  to make sure special characters are handled. Example: SHIFT + 3 = #
+- The utf8 result from above is printed to STDOUT (flushed at after each character).
+- Source code: https://github.com/SohamG/door-driver
```
Comment (clee231) [inline docs/kb/office-door.md:23]: Do you think we'll ever move this over to the lugatuic organization?

File: docs/kb/office-door.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+### Shell script
+
+- This is the legacy card_reader.sh (now called `new_card_reader.sh` script modified to have `?` as the delimiter.
```
Comment (clee231) [inline docs/kb/office-door.md:27]: A PR for this new change should be made to the existing repo at https://github.com/lugatuic/doorkeeper-controller.

File: docs/kb/office-door.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+- UIC ID Cards have ? as the last character.
+- The output of the EVDEV Driver is piped to this shell script.
+- This script contains a list of UINs allowed to open the door.
+- TODO: Make it read allowed UINs from a file (BASH `mapfile` builtin).
+- If the card is of an allowed UIN, the python scirpt to open the door is called.
```
Comment (clee231) [inline docs/kb/office-door.md:32]:
```suggestion
- If the card is of an allowed UIN, the python script to open the door is called.
```
Small typo.

File: docs/kb/office-door.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+### Deployment Information
+
+- Systemd user service called `doorkeeper.service` located in `/home/pi/.config/systemd/user/doorkeeper.service`
```
Comment (clee231) [inline docs/kb/office-door.md:36]: TODO: We should commit this systemd service to the project repo.

File: docs/kb/office-door.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+- Key press and unpress events are processed with [libxkbcommon](https://xkbcommon.org/)
+  to make sure special characters are handled. Example: SHIFT + 3 = #
+- The utf8 result from above is printed to STDOUT (flushed at after each character).
+- Source code: https://github.com/SohamG/door-driver
+
+### Shell script
+
+- This is the legacy card_reader.sh (now called `new_card_reader.sh` script modified to have `?` as the delimiter.
+- UIC ID Cards have ? as the last character.
+- The output of the EVDEV Driver is piped to this shell script.
```
Comment (clee231) [inline docs/kb/office-door.md:29]:
```suggestion
- The output of the EVDEV driver is piped to this shell script.
```
Capitalization nitpick.

File: docs/kb/office-door.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+- This script contains a list of UINs allowed to open the door.
+- TODO: Make it read allowed UINs from a file (BASH `mapfile` builtin).
+- If the card is of an allowed UIN, the python scirpt to open the door is called.
+
+### Deployment Information
+
+- Systemd user service called `doorkeeper.service` located in `/home/pi/.config/systemd/user/doorkeeper.service`
+- This service calls a shell script called `fuck-systemd.sh` located in `/home/pi/DOOR/door-driver/`
+- This shell script calls the EVDEV Driver with its output piped to `new_card_reader.sh`
```
Comment (clee231) [inline docs/kb/office-door.md:38]:
```suggestion
- This shell script calls the EVDEV driver with its output piped to `new_card_reader.sh`
```
Capitalization nitpick.

File: docs/kb/office-door.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+- Systemd messes up I/O redirection so the intermediary script is required.
+
+## Discord Bot
+
+- Systemd system service called discordbot.service.
```
Comment (clee231) [inline docs/kb/office-door.md:43]:
```suggestion
- Systemd system service called `discordbot.service`.
```
Formatting nitpick.

File: docs/kb/office-door.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
+# ACM/LUG Office (SELE 2264) Door Documentation
+
+## Components
+
+1. MagStripe Reader for reading UIC ID Cards.
```
Comment (clee231) [inline docs/kb/office-door.md:5]:
```suggestion
1. MagStripe reader for reading UIC ID cards.
```
Capitalization suggestion.

File: docs/kb/office-door.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+   - [EVDEV](https://en.wikipedia.org/wiki/Evdev) driver written in Rust
+     to capture events directly from card reader.
+   - Shell script that parses the input and finds and compares the UIN
+   - Python script that actually uses GPIO to physically open the door.
+2. Discord bot to open the door from discord
```
Comment (clee231) [inline docs/kb/office-door.md:10]:
```suggestion
2. Discord bot to open the door from Discord
```
Grammar Nitpick: Proper nouns should be capitalized.

File: docs/kb/office-door.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+   - Work in Progress
+   - Basically just calls the python script from before.
+
+## EVDEV driver
+
+- The card reader is basically the keyboard, presented in /dev/input/eventX.
```
Comment (clee231) [inline docs/kb/office-door.md:16]:
```suggestion
- The card reader is basically the keyboard, presented in `/dev/input/eventX`.
```
Formatting suggestion.

File: docs/kb/office-door.md (hunk 4 above)
Comment (clee231) [inline docs/kb/office-door.md:27]:
```suggestion
- This is the legacy `card_reader.sh` (now called `new_card_reader.sh` script modified to have `?` as the delimiter.
```
Formatting suggestion.

File: docs/kb/office-door.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  the input to also go there).
+- Key press and unpress events are processed with [libxkbcommon](https://xkbcommon.org/)
+  to make sure special characters are handled. Example: SHIFT + 3 = #
+- The utf8 result from above is printed to STDOUT (flushed at after each character).
+- Source code: https://github.com/SohamG/door-driver
+
+### Shell script
+
+- This is the legacy card_reader.sh (now called `new_card_reader.sh` script modified to have `?` as the delimiter.
+- UIC ID Cards have ? as the last character.
```
Comment (clee231) [inline docs/kb/office-door.md:28]:
```suggestion
- UIC ID cards have ? as the last character.
```
Capitalization suggestion.

**Review by clee231 — CHANGES_REQUESTED** [review]

File: docs/kb/office-door.md (hunk 3 above)
Comment (SohamG) [reply docs/kb/office-door.md:23]: Uh yeah I accidentally created the repo in my personal account. I'll get it over to lugatuic

File: docs/kb/office-door.md (hunk 4 above)
Comment (SohamG) [reply docs/kb/office-door.md:27]: Sure. I think I should probably make that repo a submodule in the driver repo

**Comment by SohamG** [comment]
UPDATE: Made new repo https://github.com/lugatuic/doorkeeper-driver as a meta repo (kinda). Tracking project specific stuff there. LMK if theres anything else other than the issues on that repo. 

PS I will be squashing once i have the :white_check_mark:

File: docs/kb/office-door.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+# ACM/LUG Office (SELE 2264) Door Documentation
+
+## Components
+
+1. MagStripe Reader for reading UIC ID Cards.
+   - [EVDEV](https://en.wikipedia.org/wiki/Evdev) driver written in Rust
+     to capture events directly from card reader.
+   - Shell script that parses the input and finds and compares the UIN
+   - Python script that actually uses GPIO to physically open the door.
+   - Repo: https://github.com/lugatuic/doorkeeper-driver
```
Comment (bmiddha) [inline docs/kb/office-door.md:10]:
```suggestion
   - Repository: [lugatuic/doorkeeper-driver](https://github.com/lugatuic/doorkeeper-driver)
```

nit: avoid bare links. and when using bare links wrap them in `<link>`

**Review by clee231 — CHANGES_REQUESTED** [review]
Sounds like PR checks are failing because you need to run `prettier`?

**Pull request closed without merging**